api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirPractitionerServiceImpl.java

```diff
@@ -92,17 +97,36 @@ public Practitioner create(@Nonnull Practitioner newResource) {
 	@Override
 	@Transactional(readOnly = true)
 	public IBundleProvider searchForPractitioners(PractitionerSearchParams practitionerSearchParams) {
-		IBundleProvider providerBundle = searchQuery.getQueryResults(practitionerSearchParams.toSearchParameterMap(), dao,
-		    translator, searchQueryInclude);
-		SearchParameterMap theParams = new SearchParameterMap();
-		IBundleProvider userBundle = userService.searchForUsers(theParams);
 		
-		if (!providerBundle.isEmpty() && !userBundle.isEmpty()) {
+		IBundleProvider providerBundle = null;
+		IBundleProvider userBundle = null;
+		
+		if (shouldSearchExplicitlyFor(practitionerSearchParams.getTag(), "provider")) {
+			providerBundle = searchQuery.getQueryResults(practitionerSearchParams.toSearchParameterMap(), dao, translator,
+			    searchQueryInclude);
+		}
+		
+		if (shouldSearchExplicitlyFor(practitionerSearchParams.getTag(), "user")) {
+			SearchParameterMap theParams = new SearchParameterMap();
+			userBundle = userService.searchForUsers(theParams);
+		}
+		
+		if (providerBundle != null && userBundle != null) {
 			return new TwoSearchQueryBundleProvider(providerBundle, userBundle, globalPropertyService);
-		} else if (providerBundle.isEmpty() && !userBundle.isEmpty()) {
+		} else if (providerBundle == null && userBundle != null) {
```

Comment:
Not sure if I should still test for "isEmpty()" here, but the Encounter search example I am following didn't.